### ISSUE TITLE
Multiple Issues and Suggestions

### ISSUE THREAD

**Issue by imsamuka** [cooperhammond/irs#82]
I know, it's better to make singular issues, but there's too many. If necessary, some of them should be externalized to other separated issues.

### Issues

* In `--select` there's a offset somewhere (i.e. selecting 1 actually selects 2).
* In `--album` there's no way to skip a song: or you download it or close the whole program and download everything manually.
* `--url` doesn't work. The song and metadata is found correctly, but the URL is never "valid". Maybe something changed a while ago.
* The program softlocks sometimes (probably because of `-S`)
* Artists with japanese names can't be found on spotify

* It should be a way to overwrite `search_terms` on runtime too (i.e. using `-t, --search-terms`), since it's very useful to retry the query using different terms, when `-S` only shows bad results.

### Setup Process

* Setting `IRS_CONFIG_LOCATION` should be **optional**: It's more safe, convenient, and consistent to select a default location for your configuration file. I suggest `$HOME/.config/irs.yaml` or `$HOME/.config/irs/config.yaml` for UNIX systems. Creating dotfiles (and dotfolders?) directly on `$HOME` is a bad idea, there's already too many applications that don't respect our `$HOME`!
* By having a default location for your configuration file, you can simplify the setup process: you can copy the _default_ config file there, and after a `--install`, the program runs out of the box!
* `--config` is a bit useless. Usually, the `-c` option on is used to provide a **new** configuration file to use.

I think this summarizes everything. I loved the application, but really there's too many problems here. I can't help much rn, because i don't know Crystal, but when I learn, I will fix these personally if they weren't fixed. Thanks for the job!

**Comment by imsamuka** [comment]
I fixed most of the issues on #83. There's some notes:

> The program softlocks sometimes (probably because of -S)

It softlocks when `youtube-dl` fails. It's probably a good idea to retry the download some times or at least close the program and warn the user when it happens.

> Artists with japanese names can't be found on spotify

On my PR, some japanese names still can't be found, but i think that's probably because of some perk on the API. After fixing the URL encodings most of these problems were fixed.
